distributed/scheduler.py

```diff
@@ -3652,31 +3650,40 @@ async def add_worker(
             except Exception as e:
                 logger.exception(e)
 
-        recommendations: dict = {}
         client_msgs: dict = {}
         worker_msgs: dict = {}
         if nbytes:
             assert isinstance(nbytes, dict)
             already_released_keys = []
             for key in nbytes:
-                ts: TaskState = self.tasks.get(key)  # type: ignore
+                ts: TaskState | None = self.tasks.get(key)
                 if ts is not None and ts.state != "released":
                     if ts.state == "memory":
                         self.add_keys(worker=address, keys=[key])
                     else:
-                        t: tuple = self._transition(
+                        recommendations, new_cmsgs, new_wmsgs = self._transition(
                             key,
                             "memory",
                             stimulus_id,
                             worker=address,
                             nbytes=nbytes[key],
                             typename=types[key],
                         )
-                        recommendations, client_msgs, worker_msgs = t
```

Comment:
This is @fjetter's key fix. Notice we were overwriting `client_msgs, worker_msgs` on each iteration.